zerver/lib/actions.py

```diff
@@ -2042,6 +2052,33 @@ def get_default_subs(user_profile):
     # to some day further customize how we set up default streams for new users.
     return get_default_streams_for_realm(user_profile.realm)
 
+#returns only default streams
+def do_get_default_streams_for_realm(realm):
+    # type: (Realm) -> List[Stream]
+    query = get_occupied_streams(realm)
+
+    def make_dict(row):
+        return dict(
+            stream_id = row.id,
+            name = row.name,
+            description = row.description,
+            invite_only = row.invite_only,
+        )
+
+    streams = [make_dict(row) for row in query]
+    streams.sort(key=lambda elt: elt["name"])
+    is_default = {}
+    default_streams = []
+    default_streams_objects = get_default_streams_for_realm(realm)
+    for stream in default_streams_objects:
+        is_default[stream.id] = True
+    for stream in streams:
+        if(is_default.get(stream["stream_id"], False)):
+            default_streams.append(stream)
+
+    return default_streams
```

Comment:
Why can't you just use something like `[default_stream.stream for default_stream in get_default_streams_in_realm(realm)]` instead of this function?